src/odemis/driver/smaract.py

```diff
@@ -4913,6 +4922,406 @@ def _cancelReference(self, future):
+            for ch, num in self._channels.items():
+                self.referenced._value[ch] = bool(si.GetProperty_i32(self.id, si.EPK(ps.Property.CH_IS_VALID, num, 0)))
+            # If the measuring system is stable and ready to produce accurate results start the polling
+            if bool(si.GetProperty_i32(self.id, si.EPK(ps.Property.SYS_IS_STABLE, 0, 0))):
+                # Start polling thread
+                self._polling_thread = util.RepeatingTimer(1, self._updatePosition, "Position polling")
+                self._polling_thread.start()
+                self._updatePosition()
+            else:
+                raise ValueError("The system did not become stable after the referencing procedure.")
+            # We only notify after updating the position so that when a listener
+            # receives updates both values are already updated.
+            self.referenced.notify(self.referenced.value)
+
+            # The adjustment state should be "disabled" after automatic adjustment. If the referencing
+            # procedure was stopped prematurely, right after it was set to "manual", it might be in the
+            # wrong state --> make sure it's set to disabled here.
+            if si.GetProperty_i32(self.id, si.EPK(ps.Property.AF_ADJUSTMENT_STATE, 0, 0)) != ps.AdjustmentState.DISABLED:
+                si.SetProperty_i32(self.id, si.EPK(ps.Property.AF_ADJUSTMENT_STATE, 0, 0), ps.AdjustmentState.DISABLED)
+                self._wait_for_progress_event(ps.AdjustmentState.DISABLED, timeout=600)  # state will be DISABLED when done
+            si.SetProperty_i32(self.id, si.Property.EVENT_NOTIFICATION_ENABLED << 16 | ps.EventType.AF_ADJUSTMENT_PROGRESS, si.DISABLED)
+
+    def _getReadableSIError(self, error):
+        """
+        Get readable SI error for logging purposes.
+
+        :param error: (si.Error or ps.Error) The error/exception raised.
+        :return: (str) The readable SI error.
+
+        """
+        err_name = "(0x{:04X})".format(error.code)
+        if error.code in set(err.value for err in si.ErrorCode):
+            err_name = "si.ErrorCode." + si.ErrorCode(error.code).name + " " + err_name
+        elif error.code in set(err.value for err in ps.ErrorCode):
+            err_name = "ps.ErrorCode." + ps.ErrorCode(error.code).name + " " + err_name
+        return f"SI function {error.func} raised error: {err_name}"
+
+    def _getReadableEvent(self, event):
+        """
+        Get readable event name for logging purposes.
+
+        :param error: (si.EventType or ps.EventType) The event.
+        :return: (str) The readable event name.
+
+        """
+        ev_name = "(0x{:04X})".format(event.type)
+        if event.type in set(ev.value for ev in si.EventType):
+            ev_name = "si.EventType." + si.EventType(event.type).name + " " + ev_name
+        elif event.type in set(ev.value for ev in ps.EventType):
+            ev_name = "ps.EventType." + ps.EventType(event.type).name + " " + ev_name
```

Comment:
For this sort of code, I think it's better to think of it in an "optimistic" way. Try to get the si.EventType, and let python find out if it exists or not. If not, it will raise an exception. In this end, it's less code executed.
Something like that (and you could update `getReadableError` similarly.
```suggestion

        try:
            ev_name = "si.EventType." + si.EventType(event.type).name + " " + ev_name
        except ValueError:
            try:
                  ev_name = "ps.EventType." + ps.EventType(event.type).name + " " + ev_name
            except ValueError:
                  ev_name = "(0x{:04X})".format(event.type)

```